src/lib/Persistence/Legacy/Content/TreeHandler.php

```diff
@@ -53,25 +53,31 @@ class TreeHandler
      */
     protected $fieldHandler;
 
+    /** @var iterable<\Ibexa\Contracts\Core\Persistence\Content\DeleteContentHandler> */
+    private iterable $deleteContentHandlers;
+
     /**
      * @param \Ibexa\Core\Persistence\Legacy\Content\Location\Gateway $locationGateway
      * @param \Ibexa\Core\Persistence\Legacy\Content\Location\Mapper $locationMapper
      * @param \Ibexa\Core\Persistence\Legacy\Content\Gateway $contentGateway
      * @param \Ibexa\Core\Persistence\Legacy\Content\Mapper $contentMapper
      * @param \Ibexa\Core\Persistence\Legacy\Content\FieldHandler $fieldHandler
+     * @param iterable<\Ibexa\Contracts\Core\Persistence\Content\DeleteContentHandler> $deleteContentHandlers
      */
     public function __construct(
         LocationGateway $locationGateway,
         LocationMapper $locationMapper,
         ContentGateway $contentGateway,
         ContentMapper $contentMapper,
-        FieldHandler $fieldHandler
+        FieldHandler $fieldHandler,
+        iterable $deleteContentHandlers,
```

Comment:
PHP 8 code detected ;)
```suggestion
        iterable $deleteContentHandlers
```